src/Http/Http.Results/src/FileResult.cs

```diff
@@ -1,86 +1,62 @@
 // Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 
-using System.Diagnostics.CodeAnalysis;
+using Microsoft.AspNetCore.Internal;
 using Microsoft.Extensions.Logging;
 using Microsoft.Net.Http.Headers;
 
 namespace Microsoft.AspNetCore.Http.Result;
 
-internal abstract partial class FileResult
+internal abstract class FileResult : FileResultBase, IResult
 {
-    private string? _fileDownloadName;
-
-    /// <summary>
-    /// Creates a new <see cref="FileResult"/> instance with
-    /// the provided <paramref name="contentType"/>.
-    /// </summary>
-    /// <param name="contentType">The Content-Type header of the response.</param>
-    protected FileResult(string? contentType)
+    public FileResult(string? contentType)
+        : base(contentType)
     {
-        ContentType = contentType ?? "application/octet-stream";
     }
 
-    /// <summary>
-    /// Gets the Content-Type header for the response.
-    /// </summary>
-    public string ContentType { get; }
+    protected abstract ILogger GetLogger(HttpContext httpContext);
 
-    /// <summary>
-    /// Gets the file name that will be used in the Content-Disposition header of the response.
-    /// </summary>
-    [AllowNull]
-    public string FileDownloadName
-    {
-        get { return _fileDownloadName ?? string.Empty; }
-        init { _fileDownloadName = value; }
-    }
+    protected abstract Task ExecuteAsync(HttpContext httpContext, RangeItemHeaderValue? range, long rangeLength);
 
-    /// <summary>
-    /// Gets or sets the last modified information associated with the <see cref="FileResult"/>.
-    /// </summary>
-    public DateTimeOffset? LastModified { get; init; }
-
-    /// <summary>
-    /// Gets or sets the etag associated with the <see cref="FileResult"/>.
-    /// </summary>
-    public EntityTagHeaderValue? EntityTag { get; init; }
+    public virtual Task ExecuteAsync(HttpContext httpContext)
+    {
+        var logger = GetLogger(httpContext);
```

Comment:
```suggestion
        var logger = httpContext.RequestServices.GetRequiredService(typeof(ILogger<>).MakeGenericType(GetType())) as ILogger;
```
How about creating `ILogger` like this so there's no need to override `GetLogger` in every class?

Reply:
MakeGenericType isn't linker friendly, though these are classes. I'd like to keep using the generic path.

Reply:
Couldn't you alternatively do `loggerFactory.GetLogger(GetType())` in the ctor? It's kinda weird to have a `GetLogger` method just to get to the implementing type.

Reply:
It can't be in the ctor, there's no way to get the logger there.